### PR TITLE
Add linting and tests

### PR DESCRIPTION
**Pull request by annekainicUSDS**
In the process of working on moving the routing functionality into this repo, I realized the need for having linting and testing in this repo. Instead of adding it to my existing branch, I wanted to push up a separate branch for this to make it easier to review.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    'padded-blocks': 0,
+    'jsx-a11y/no-static-element-interactions': 0,
+    'import/prefer-default-export': 0,
+    'global-require': 0,
+    'no-lonely-if': 0,
+    'lines-around-directive': 0,
+    'jsx-a11y/href-no-hash': 0,
+  },
+  'overrides': [{
+    'files': ["**/*.spec.jsx", "**/*.spec.js", "src/platform/testing/**/*.js", "src/platform/testing/**/*.jsx"],
```
[the remark this answers is not in the document]
Comment (annekainicUSDS) [reply .eslintrc]: Ah good catch! Copied this from Vets.gov as a starting point and missed this.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    'max-len': [0],  // No agreed upon max length and react lines get > 100 chars easily.
+    'arrow-parens': 0, // We don't have an agreed upon pattern here
+    'no-underscore-dangle': 0, // We have build flags that use this
+     // Disabled rules with rationale.
+    'react/no-multi-comp': 0,      // Leave organization to code reviewer discretion.
+    'react/prefer-stateless-function': 0, // Leave statelessness to code reviewer discretion.
+     'react/jsx-closing-bracket-location': [2, 'after-props'],
+    'react/jsx-equals-spacing': 2,
+    'react/jsx-key': 2,
+    'react/jsx-no-bind': [0, {'ignoreRefs': true}],  // TODO: Enable after fixing.
```
[the remark this answers is not in the document]
Comment (annekainicUSDS) [reply .eslintrc]: Will double check these!

**Comment by dmethvin-gov** [comment]
I agree that the alphabetical part seems unnecessarily strict. Enforcing prop-types seems like a good idea if we're going to use prop-types at all ... how broken are we right now with respect to this? 😄 If it could be fixed with a few updates great. If lots of files need fixing, maybe just leave that one with a comment and we can do it later?

**Comment by annekainicUSDS** [comment]
@dmethvin-gov I'm going to keep the alphabetical linters off for now, and I enabled the prop types one and added prop types. I wasn't too bad! I think we should try enabling this and see if it ever becomes too burdensome, but it seems like it could be useful. Let me know if you agree!

**Comment by annekainicUSDS** [comment]
@dmethvin-gov I updated the eslint config to match the version that's now in USFS based on your changes. Let me know how this looks to you!

**Pull request merged**